tests/NRedisStack.Tests/Json/JsonTests.cs

```diff
@@ -103,10 +101,31 @@ public void TestSetFromDirectory()
     [Fact]
     public void TestJsonSetNotExist()
     {
+        //arrange
+        var conn = redisFixture.Redis;
+        var db = conn.GetDatabase();
+        IJsonCommands commands = new JsonCommands(db);
+        db.Execute("FLUSHALL");
+
+        var obj = new Person { Name = "Shachar", Age = 23 };
+        Assert.True(commands.Set("Person:Shachar", "$", obj, When.NotExists));
+        Assert.False(commands.Set("Person:Shachar", "$", obj, When.NotExists));
+        Assert.True(commands.Set("Person:Shachar", "$", obj, When.Exists));
+    }
+
+    [Fact]
+    public async Task TestJsonSetNotExistAsync()
+    {
+        //arrange
+        var conn = redisFixture.Redis;
+        var db = conn.GetDatabase();
+        IJsonCommandsAsync commands = new JsonCommands(db);
```

Comment:
Inconsistent use of `var`/type declaration. Same as above.

Reply:
If we use var then `commands` will be of type `JsonCommands` and not `IJsonCommands`.

Reply:
What about using the `JsonCommandsAsync` constructor instead of casting?

Reply:
I just tested locally and it works properly even when using var, I changed it in all the tests in `jsonTests.cs`.
Now it is consistent. What do you think?

Reply:
Looks good!
As the implementation is (almost) based upon the same approach as `StackExchange.Redis`, where the `*Commands` inherits from `*CommandsAsync`, I do not have a strong opinion on using one or the other as long as the method matches the scenario 👍 

One could also use the extension methods to get the same result, i.e., `db.JSON()`.